File: Lasmart.Web/Services/PointService.cs
using Lasmart.Web.Models;
using Lasmart.Web.Services.IServices;

namespace Lasmart.Web.Services;

public class PointService : BaseService, IPointService
{
    public PointService(IHttpClientFactory clientFactory) : base(clientFactory)
    { }

    public async Task<ResponseDto> GetPointsAsync()
    {
        return await this.SendAsync(new ApiRequest
        {
            ApiMethod = SD.ApiMethod.GET,
            Url = SD.PointAPIBase + "api/points"
        });
    }

    public async Task<ResponseDto> DeletePointAsync(int pointId)
    {
        return await this.SendAsync(new ApiRequest
        {
            ApiMethod = SD.ApiMethod.GET,
            Url = SD.PointAPIBase + $"api/points/pointId={pointId}"
        });
    }
}

File: Lasmart.Web/Scripts/Services/PointService.ts
import {IPointService} from "./IServices/IPointService.js";
import {BaseService} from "./BaseService.js";
import {ResponseDto} from "../Models/ResponseDto.js";
import {ApiMethod} from "../Models/ApiMethod.js";
import {ApiRequest} from "../Models/ApiRequest.js";
export class PointService extends BaseService implements IPointService {
    
    constructor(baseUrl: string)  {
        super(baseUrl)
    }
    deletePointByIdAsync(pointId: number): Promise<ResponseDto> {
        return this.sendAsync(new ApiRequest(
            ApiMethod.DELETE,
            `api/points/pointId=${pointId}`
        ));
    }

    getPointsAsync(): Promise<ResponseDto> {
        return this.sendAsync(new ApiRequest(
            ApiMethod.GET,
            "api/points"
        ));
    }
    
}